### DISCUSSION TITLE
Adding STUN/TURN/ICE

### DISCUSSION THREAD

**Discussion by roman-dzieciol** [microsoft/msquic#3590] [General]
Hello,

I'd like to add STUN/TURN/ICE support to a fork of MsQuic in context of P2P gaming. In particular it will need to support those protocols on the same socket, i.e. handle TURN Data Channel relay packet frames.

What changes would you recommend for that?

This PR looks useful: https://github.com/microsoft/msquic/pull/2299

Thanks!
Roman

**Comment by nibanks** [comment]
Hi @roman-dzieciol, thanks for the post! Would you mind [submitting an official feature request](https://github.com/microsoft/msquic/issues/new?assignees=&labels=feature+request&template=feature_request.yaml&title=Feature+request+title) with this info an all the details/background/requirements you have? We've had this question come up a couple times but no one seemed to need it enough to follow through, so we've not prioritized it yet. Thanks!
